### PR TITLE
update service endpoint section

### PR DESCRIPTION
**Pull request by LiranCohen**
First pass at aligning the `Service Endpoints` sections.

When looking at this I found myself asking a question which I figured will be good to document here and talk on the call.

Does the `enc` or `sig` key have to reference a key that is associated with the resolved `DID`?

For example could the JSON example shown have `"sig": ["did:eample:bob#dwn-sig"]` ?

### CODE REVIEW

**Comment by csuwildcat** [comment]
LMAO, I just did a PR for this too.

**Comment by andorsk** [comment]
Note: do not follow beyond 1 levels deep recursion for `did` resolution.

**Comment by andorsk** [comment]
@andorsk made a comment. TODO: add language around using service-profile for dwn spec.

**Comment by LiranCohen** [comment]
Added the wording regarding recursion.

@andorsk Let's create a separate issue for the service-profile so that we can better define what properties can exist in the service profile. Could you also include a link to the service profile spec/docs in that issue?

File: spec/spec.md [hunk 1]
```diff
@@ -137,9 +137,37 @@ Object Signing / Encryption |
...
+    ]
+  }]
+}
+```
+
+- Service Endpoints ****MUST**** contain an `id` property and it ****MUST**** be set to `#dwn`.
+- Service Endpoints ****MUST**** contain a `type` property and it ****MUST**** be set to `DecentralizedWebNode`.
+- Service Endpoints ****MUST**** contain a `serviceEndpoint` property and it ****MAY**** be set to:
+  - either a `string` value which represents the `URL` associated with the `DWN`
+  - an array of `string` values which each value represents a `URL` associated with a `DWN`
```
Comment (TallTed) [inline spec/spec.md:166]: Is this really meant to be a MAY? Or is it meant as I've adjusted below?
```suggestion
- Service Endpoints ****MUST**** contain a `serviceEndpoint` property and it ****MUST**** be set to one of the following:
  - a `string` value which represents the `URL` associated with the `DWN`
  - an array of `string` values, each of which represents the `URL` associated with a `DWN`
```

File: spec/spec.md [hunk 2]
```diff
@@ -137,9 +137,37 @@ Object Signing / Encryption |
...
+- If a Service Endpoint `URL` is a `DID`, it ****MUST NOT**** not be followed more than 1 level deep when resolving.
+- Service Endpoints ****MUST**** contain an `enc` property and it ****MUST**** be a `string` value or an array of `string` values which represent the `keyId` associated with `encrypting` data.
```
Comment (TallTed) [inline spec/spec.md:168]:
```suggestion
- Service Endpoints ****MUST**** contain an `enc` property which ****MUST**** be a `string` value or an array of `string` values, each of which represents a `keyId` associated with `encrypting` data.
```

File: spec/spec.md [hunk 3]
```diff
@@ -137,9 +137,37 @@ Object Signing / Encryption |
...
+- Service Endpoints ****MUST**** contain a `sig`property and it ****MUST**** be a `string` value or an array of `string` values which represent the `keyId` associated with `signing` data.
```
Comment (TallTed) [inline spec/spec.md:169]:
```suggestion
- Service Endpoints ****MUST**** contain a `sig` property which ****MUST**** be a `string` value or an array of `string` values, each of which represents a `keyId` associated with `signing` data.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by andorsk** [comment]
Can we merge @TallTed 's updates and get this in?

**Comment by LiranCohen** [comment]
Thank you @TallTed !! As always, excellent work!

**Review by andorsk — APPROVED** [review]
after TallTed's updates, I think this looks good to go.

**Comment by LiranCohen** [comment]
Addresses: https://github.com/decentralized-identity/decentralized-web-node/issues/284

**Pull request merged**